docs/Architecture.md

```diff
@@ -0,0 +1,502 @@
+
+### Babylon Side
+
+On the babylon side, the main module is the Babylon `btcstaking` module, which
+provides the functionality of Babylon Staking integration. It is responsible
+for managing the staking and unstaking of $BTC, as well as providing the
+finality providers with the necessary information to operate.
+There's also a `btcstkconsumer` module, which is responsible for managing the
+staking part that is specific to the Consumer chains.
+The `btcstkconsumer` module interacts with the `zoneconcierge` module, which is
+responsible for managing the restaking of $BTC on the Bitcoin Supercharged
+Networks (BSNs).
+The BSN is a network of Cosmos chains that are integrated with Babylon Staking,
+and receive economic security from Bitcoin.
+
+### BSN Side
+
+On the BSN side, the main contract is the `btc-staking` contract, which is
+responsible for managing the staking and unstaking of $BTC on the BSN. It has
+the necessary information related to delegations of Bitcoin on the BSN.
+The `btc-staking` contract interacts with Babylon through the `babylon`
+smart contract, which is responsible for managing the communication between
+Babylon and the BSN over IBC. The `babylon-contract` contract is responsible for
+handling the IBC messages and relaying them to the appropriate contracts on the
+BSN side.
+It also deals with contract's instantiation, and with storing the information of
+a transfer channel between the BSN and Babylon, for opt-in rewards transfer in
+the form of the BSN's native token, to Babylon.
+
+Another important contract on the BSN side is the `btc-finality` contract,
+which handles the information about signatures and public randomness sent by
+the finality providers operating on the BSN. It also deals with vote tallying,
+and defines the finalisation status of the blocks on the BSN.
+
+Finally, the `btc-light-client` contract is responsible for maintaining the
+light client state of the Bitcoin network on the BSN. It is used to verify
+the delegations and timestamping sent through Babylon, and to ensure that the
+BSN is in sync with the Bitcoin network.
+
+## Detailed Interfaces
+
+This section provides a detailed overview of the interfaces provided by the
+different contracts involved in the Babylon Staking integration.
+It only describes the interfaces that are relevant for the Babylon Staking
+integration, and does not cover the full functionality of the contracts. It also
+does not cover the different queries that the contracts provide, which can be
+found in the contract's documentation.
+
+### Messages
+
+There are four main message types that are used in the Babylon Staking
+integration.
+
+- IBC messages, which are used to communicate between the Babylon and BSN sides.
+- Execute messages, which are used to execute actions on the contracts.
+- Sudo messages, which are used by the `babylon` module which is part of
+  babylon-sdk to interact with the Babylon contracts; typically during begin /
+  end block processing.
```

Comment:
I think the documentation relevant to Babylon SDK can be put under Babylon SDK repository, and then in the contracts' doc we can just refer to the doc in Babylon SDK.